### ISSUE TITLE
Reinitialize & ReinitializeWithUpload with SqlSyncChangeTrackingProvider

### ISSUE THREAD

**Issue by safaairaq100** [Mimetis/Dotmim.Sync#1188]
Hello, thank you for this wonderful work.
I have done several tests on this framework and it was very good, but I faced a problem that I wanted to test but could not, which is if the client stopped synchronizing for a long time and added several records  **without tracking.** After that, I synchronized it with the server using **_ReinitializeWithUpload_**, but I faced a problem that the table does not contain a trigger. I use **_SqlSyncChangeTrackingProvider_** and this does not need triggers. This problem occurs in **_Reinitialize_** and **_ReinitializeWithUpload_** only and does not occur with the **_Normal_** option. Did I miss something or a specific setting that I should do?

![WhatsApp Image 2024-06-12 at 12 29 58_16596a05](https://github.com/Mimetis/Dotmim.Sync/assets/18406385/5fc34a5c-0e23-4882-bcca-b354f06beab0)
![WhatsApp Image 2024-06-12 at 13 06 41_a7ffdfc4](https://github.com/Mimetis/Dotmim.Sync/assets/18406385/67ed79a5-e073-4acd-88af-30b7c219c00e)

Client: Console Application
SQL SERVER 2019
 `
 
 class Program
    {
    

    private static string clientConnectionString = $"Data Source=DESKTOP-517L98Q\\SQL2019; " +
            $"Initial Catalog=Test2024;Integrated Security=true;TrustServerCertificate=YES;";
        static async Task Main(string[] args)
        {
            Console.WriteLine("Be sure the web API has started. Then click enter..");
            Console.ReadLine();
            await SynchronizeAsync();
        }

        private static async Task SynchronizeAsync()
        {
         
            var serverOrchestrator = new WebRemoteOrchestrator("https://localhost:44342/api/sync");
            var clientProvider = new SqlSyncChangeTrackingProvider(clientConnectionString);
            var options = new SyncOptions
            {
                BatchSize = 1000,
            };
            var agent = new SyncAgent(clientProvider, serverOrchestrator, options);
         
            //var clientScope = await agent.LocalOrchestrator.ProvisionAsync(await serverOrchestrator.GetScopeInfoAsync("DefaultScope"), overwrite: true);
            //await agent.LocalOrchestrator.DeprovisionAsync();
            do
            {
                try
                {
                    var progress = new SynchronousProgress<ProgressArgs>(args => Console.WriteLine($"{args.ProgressPercentage:p}:\t{args.Message}"));
                    var s1 = await agent.SynchronizeAsync(SyncType.ReinitializeWithUpload, progress);
                    Console.WriteLine(s1);
                }
                catch (Exception ex)
                {
                    Console.WriteLine(ex.Message);
                }

            } while (Console.ReadKey().Key != ConsoleKey.Escape);

            Console.WriteLine("End");
        }
    }
`

Server: Web .Net 8 Core 
SQL SERVER 2019

`public class Startup
    
    
    {
        public Startup(IConfiguration configuration) => Configuration = configuration;
        public IConfiguration Configuration { get; }
        public async void ConfigureServices(IServiceCollection services)
        {
            services.AddControllers();

            services.AddDistributedMemoryCache();
            services.AddSession(options => options.IdleTimeout = TimeSpan.FromMinutes(30));
            var connectionString = Configuration.GetSection("ConnectionStrings")["SqlConnection"];
         
            var options = new SyncOptions
            {
                SnapshotsDirectory = "C:\\Tmp\\Snapshots",
                BatchSize = 2000,
            };

            var setup = new SyncSetup("Tb_Menu", "Tb_Vou", "Tb_Bill_Body", "Tb_Bill_Footer", "Tb_Bill_Header");
     
            var provider = new SqlSyncChangeTrackingProvider(connectionString);
            services.AddSyncServer(provider, setup, options);

        }
        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            if (env.IsDevelopment())
                app.UseDeveloperExceptionPage();

            app.UseHttpsRedirection();
            app.UseRouting();
            app.UseAuthorization();
            app.UseSession();
            app.UseEndpoints(endpoints => endpoints.MapControllers());
        }
    }`

I did ProvisionAsync and DeprovisionAsync same problem.

**Comment by safaairaq100** [comment]
After doing a debugging I found the problem in the **_InternalResetTableAsync_** method that call 
   ```
 var (command, _) = await this.InternalGetCommandAsync(scopeInfo, context, syncAdapter, DbCommandType.Reset,
        runner.Connection, runner.Transaction, runner.CancellationToken, runner.Progress).ConfigureAwait(false);
```
This method generates Commands that must not be generated with **_SqlSyncChangeTrackingProvider_**

![Capture22](https://github.com/Mimetis/Dotmim.Sync/assets/18406385/222bfb98-fe51-4d17-a36e-476954399c8f)
![Capture223](https://github.com/Mimetis/Dotmim.Sync/assets/18406385/b40f5703-fcc1-4030-99ad-e522add6a1fc)

The commands that cause the problem 
```
SET @sync_row_count = 0;

DISABLE TRIGGER [dbo].[Tb_Doctors_update_trigger] ON [Tb_Doctors];
DISABLE TRIGGER [dbo].[Tb_Doctors_insert_trigger] ON [Tb_Doctors];
DISABLE TRIGGER [dbo].[Tb_Doctors_delete_trigger] ON [Tb_Doctors];
DELETE FROM [Tb_Doctors];
DELETE FROM [Tb_Doctors_tracking];  
ENABLE TRIGGER [dbo].[Tb_Doctors_update_trigger] ON [Tb_Doctors];
ENABLE TRIGGER [dbo].[Tb_Doctors_insert_trigger] ON [Tb_Doctors];
ENABLE TRIGGER [dbo].[Tb_Doctors_delete_trigger] ON [Tb_Doctors];

SET @sync_row_count = @@ROWCOUNT;
```
The  **_SqlSyncChangeTrackingProvider_**  does not create [Tb_Doctors_tracking]; or the other **TRIGGERS** table.

**Comment by safaairaq100** [comment]
I just make change in **_CreateResetCommand_** method and remove:

```
     stringBuilder.AppendLine($"DISABLE TRIGGER {updTriggerName} ON {tableName.Schema().Quoted()};");
     stringBuilder.AppendLine($"DISABLE TRIGGER {insTriggerName} ON {tableName.Schema().Quoted()};");
     stringBuilder.AppendLine($"DISABLE TRIGGER {delTriggerName} ON {tableName.Schema().Quoted()};");
     stringBuilder.AppendLine($"DELETE FROM {trackingName.Schema().Quoted()};");
     stringBuilder.AppendLine($"ENABLE TRIGGER {updTriggerName} ON {tableName.Schema().Quoted()};");
     stringBuilder.AppendLine($"ENABLE TRIGGER {insTriggerName} ON {tableName.Schema().Quoted()};");
     stringBuilder.AppendLine($"ENABLE TRIGGER {delTriggerName} ON {tableName.Schema().Quoted()};");
```

To:

  ```

 private string CreateResetCommand()
   {
  
       StringBuilder stringBuilder = new StringBuilder();
       stringBuilder.AppendLine($"SET @sync_row_count = 0;");
       stringBuilder.AppendLine();

       stringBuilder.AppendLine($"DELETE FROM {tableName.Schema().Quoted()};");

       stringBuilder.AppendLine();
       stringBuilder.AppendLine(string.Concat("SET @sync_row_count = @@ROWCOUNT;"));

       return stringBuilder.ToString();
   }
```
And everything works great :)

**Comment by Mimetis** [comment]
thanks for the detailed feedback
I'm looking into it
I'll let you know

**Comment by Mimetis** [comment]
The issue has been fixed with version `1.0.2`
Thanks for finding this bug !
